plugins/janus_videoroom.c

```diff
@@ -4369,6 +4374,34 @@ void janus_videoroom_hangup_media(janus_plugin_session *handle) {
 	janus_mutex_unlock(&sessions_mutex);
 }
 
+static void janus_videoroom_hangup_subscriber(janus_videoroom_subscriber * s) {
+	/* already hangup */
+	if (!s->feed) {
+		return;
+	}
+	/* check if the owner needs to be cleaned up */
+	if(s->pvt_id > 0 && s->room != NULL) {
+		janus_mutex_lock(&s->room->mutex);
+		janus_videoroom_publisher *owner = g_hash_table_lookup(s->room->private_ids, GUINT_TO_POINTER(s->pvt_id));
+		if(owner != NULL) {
+			janus_mutex_lock(&owner->subscribers_mutex);
```

Comment:
Why lock inside the method just for this, when we could lock the whole method when we call it? (see below for related comment). From a quick glance it doesn't look we have anything that would deadlock because of that, even though I haven't checked in detail yet.